ssh/ssh_sed_unlock.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+
+trap '' 2
+
+echo
+while [ true ] ; do
+
+    if /usr/local/sbin/sedunlocksrv/opal-functions.sh | grep -q "Could not detect TCG Opal 2-compliant disks"
+    then
+        echo "Could not detect TCG Opal 2-compliant disks. Nothing to do here."
+        echo
+        echo "Press [ENTER] to shutdown"
+        while [ true ] ; do
+            read -s -N 1 -t 1 key
+            if [[ $key == $'\x0a' ]];        # if input == ENTER key
+            then
+                echo "Shutting down..."
+                poweroff -nf
+            fi
+            done
+    fi
+
+    echo -n "🔑🔑🔑 Enter SED password: "
+
+    unset password;
+    while IFS= read -r -n1 -s char; do
+        case "$char" in
+        $'\0')
+            break
+            ;;
+        $'\177')
+            if [ ${#password} -gt 0 ]; then
+                echo -ne "\b \b"
+                password=${password::-1}
+            fi
+            ;;
+        *)
+            echo -n '*'
+            password+="$char"
+            ;;
+        esac
+    done
+    echo
+
+    # do not accept empty passwords
```

Comment:
Just an idea (by no means required to complete merge): since you don't allow empty passwords, I think you could make a loop which accepts passwords (so you can unlock multiple disks with different passwords) and reboot when the user just pressed enter (empty password case).

Reply:
could we change opal-functions.sh?

https://github.com/kapitainsky/sedunlocksrv-pba/blob/3a0513a3608fee895156c3926106f59a9e06db3c/sedunlocksrv/opal-functions.sh#L169-L173

it would be easiest when it returns 1 when not all disks were unlocked.

At the moment it returns 0 and just print:

echo "$unlocked_device_count of $device_count disks unlocked. Please reboot manually."

also I do not have multiple SED disks to test it well.

Reply:
I think this the most straightforward way - I do not think it will break anything in go server as there rebooting is manual.

Reply:
If you have 2 disks with different passwords, and you unlocked the fist one. Then while/after unlocking the second one I'm not sure if unlocked_device_count is 1 or 2. It could be 1 because it only unlocked the 2nd disk in this 'iteration'. Or it could be 2 if opal_device_unlock also reruns 0 for already unlock disks... I don't remember how this works exactly and I probably had a reason to make reboot a manual action XD

Maybe instruct the user to press the Escape key for reboot via SSH? Then reboot is easy, manual, manual and doesn't 'conflict' with the enter key?

Reply:
will wait for your opinion

Reply:
let me think about it - maybe some idea will come

Reply:
just tested - I can detect ESC press - nice

So it can be used for reboot. Will implement it

Reply:
ESC - reboot
CTRL-D - shutdown

so now user can keep unlocking disks with different passwords

![Screenshot_2023-04-29_09-15-06](https://user-images.githubusercontent.com/15635386/235293416-6fabc066-c056-405c-aacb-28546c3ef368.png)

I updated README accordingly.

I am happy now I can use SSH to unlock my server + maybe others will find it handy too

Have a nice weekend!